### ISSUE TITLE
Give your image a border width of 10px.

### ISSUE THREAD

**Issue by AligatorAsteroid** [FreeCodeCamp/FreeCodeCamp#3204]
Challenge http://freecodecamp.com/challenges/waypoint-add-borders-around-your-elements has an issue. "Give your image a border width of 10px doesn't work. 

<img src='http://i.imgur.com/TcZ6gv7.png'>

**Comment by benschenker** [comment]
Hey @AligatorAsteroid thanks for posting this!

Thanks for including the screenshot.  Unfortunately, I cannot replicate your issue on my computer (it works perfectly for me).  Can you try to see if it works with a [Hard Refresh](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Refresh-Your-Browser-Cache) or in an incognito window?

**Comment by AligatorAsteroid** [comment]
A Hard Refresh or an incognito window don't work, but when I switched the browser it worked.

**Comment by awesomeaniruddh** [comment]
Which browser were you using previously?

**Comment by bearbox** [comment]
Same issue, on Chrome.

**Comment by AligatorAsteroid** [comment]
I used firefox, but it works on Edge

**Comment by srhoades** [comment]
Worked in Chrome would not work in Firefox.

**Comment by jakeekaj** [comment]
Doesnt work in Chrome for me but works in IE

**Comment by moxilla3** [comment]
Does not work for me on chrome, firefox or opera for me, cannot move forward?

**Comment by moxilla3** [comment]
<img width="985" alt="screen shot 2015-09-12 at 11 28 28 am" src="https://cloud.githubusercontent.com/assets/10538501/9833139/c56a95ae-5941-11e5-98af-8b28085493cf.png">

**Comment by 4mifix** [comment]
You have to take out the <img class="smaller-image" class="thick-green-border" src="https://bit.ly/fcc-relaxing-cat">  Wrong

<img class="smaller-image thick-green-border" src="https://bit.ly/fcc-relaxing-cat"> Right, you are styling only the img element border with class .thick-green-border{BORDER-color: green; BORDER-width: 10px; BORDER-style: solid;}

**Comment by sfilipov** [comment]
@moxilla3 In your case it doesn't work because you have not applied the class properly:
You wrote: class="smaller-image" class="thick-green-border"
You should write: class="smaller-image thick-green-border"

**Comment by Marradoren** [comment]
I have the same issue on Firefox 40.0.3. Works fine in Edge.

**Comment by thankuever** [comment]
i cant work on Chrome
but when i change to IE
it can work

**Comment by alvinphan** [comment]
Running into same issue here, using incognito on Chrome, Mac OSX.

![screen shot 2015-09-16 at 12 22 44 pm](https://cloud.githubusercontent.com/assets/12944910/9915767/b2a4a1f6-5c6d-11e5-9740-b5d6968ab2cb.png)

**Comment by tinascot** [comment]
Not working for me either. Can't move forward until this is fixed.

**Comment by yourboyvic** [comment]
incognito chrome mac did not work, trying firefox and safari brb

**Comment by yourboyvic** [comment]
![screen shot 2015-09-18 at 1 23 56 pm](https://cloud.githubusercontent.com/assets/6733145/9970195/92bbcc56-5e08-11e5-925c-4b1b1208cc55.png)

worked in Safari...sometimes this bug can be replicated sometimes it cannot, it's a good bug

**Comment by symolloy** [comment]
Same problem.  Has happened to me 3x now on different exercises. I'll copy the code from the next lesson and paste it and it still won't work.

**Comment by RikRolled** [comment]
I ran in to the same problem when i was working with Firefox. Tried it again in Edge browser and it worked without a problem.

**Comment by Jibanprakash** [comment]
its a browser compatibility issue, it works in some browsers and doesn't work in other. cheers

**Comment by nandakore** [comment]
I had this same problem, I had to change the width to 11px to pass it.

**Comment by symolloy** [comment]
Thank you! I'll try it.

**Comment by symolloy** [comment]
I'll try it. Thank you!

**Comment by tva10** [comment]
Today I founded this problem too in GoogleChrome (so I passed this task few weeks ago):
I decided to add a script using testcase from task `<script type="text/javascript">
  console.log($('img.smaller-image').css('border-top-width'));
</script>
`
to check the size and got `9.333px` because I use `scale 75%`. 

![image](https://cloud.githubusercontent.com/assets/14250295/10171902/0752ab18-66e5-11e5-859d-f15d665b8a74.png)

When I set it on `100%` it is becomes `10px`.
![image](https://cloud.githubusercontent.com/assets/14250295/10171788/85861b60-66e4-11e5-8340-5e63afbd9159.png)
When I set scale on `125%` it becomes 9.6px. 
![image](https://cloud.githubusercontent.com/assets/14250295/10171866/e5f04bba-66e4-11e5-9023-aa557cb78716.png)

**Comment by valrey** [comment]
Thanks tva10, it works now on my firefox browser :)

**Comment by fchilok** [comment]
Yea, zoom needs to be 100%, it doesn't work otherwise.

**Comment by Athenase** [comment]
Yes, I had the same problem on Chrome but then set my screen zoom to 100%. Ran the text again, and it worked.

**Comment by benschenker** [comment]
Great catch @tva10!  I was able to replicate this in chrome after setting browser zoom to 75%.

I think restoring zoom to default is definitely a workaround.  Is there anything that can be done to the test to avoid this bug altogether?

**Comment by tva10** [comment]
@benschenker so I tired to find smart solution again now. But I create another (not smart :D) using `CSSStyleDeclaration`.
I wrote function which search `selector` in all declared styles and return the last (non `undefined`) implementation of `attr`.
```
var searchSelectorAttr=function(selector, attr){
	var temp;
	for(var i=0,n=document.styleSheets.length;i<n;i++){
		if(document.styleSheets[i].cssRules === null) continue;
		for(var j=0, m=document.styleSheets[i].cssRules.length;j<m;j++){
			if(document.styleSheets[i].cssRules[j].selectorText===selector 
				&& document.styleSheets[i].cssRules[j].style.getPropertyValue(attr)){
				temp = document.styleSheets[i].cssRules[j].style.getPropertyValue(attr);
			}
		}
		//console.dir(document.styleSheets[i]);
	}
	return temp;
};
```
using: `searchSelectorAttr('.thick-green-border','border-top-width')`

some testcases for this function:
```
<style>
  .red-text {
    color: red;
  }
  /*...........*/
  .thick-green-border { 
    border-color: green;  
    border-style: solid; 
    border-width: 10px; 
  }
  .thick-green-border { 
    border-color: green;  
    border-style: solid; 
    border-width: 11px;  
    border-width: 20px; 
  }
</style>
<style>
  .thick-green-border{
  }
</style>
```
the real border in this test should be 20px, so and this `searchSelectorAttr('.thick-green-border','border-top-width')` return 20px

ps. I think I will should write a lib :DD

**Comment by xSKYx** [comment]
Thanks tva10. Good catch! Zoom out did it for me!

**Comment by catapixel** [comment]
Confirmed that running the code at 100% zoom was the only way to pass the failing test

**Comment by GregFoley** [comment]
Just spent some time on Gitter trying to figure this one out. I think I can add something: the test also failed until I changed the Windows/Display control panel setting to 100%.

**Comment by tunessofia** [comment]
I add this 
.thick-green-border{
        border-color: green; border-width: 10px;  border-style:solid;
  }
and i needed to replace this with (suggestion by one of the campers):
  .thick-green-border {
    border-color: green;
    border-width: 10px;
    border-style: solid;
   }

maybe it's something related with invisible characters apparently

**Comment by SaintPeter** [comment]
This one continues to vex.  

It's technically possible to detect browser zoom in some browsers:
http://stackoverflow.com/questions/1713771/how-to-detect-page-zoom-level-in-all-modern-browsers

It might make sense to include that as a test so at least users can tell what the issue is.

**Comment by DustInCompetent** [comment]
Sometimes the test passes, when you give the border a width of 11px .__.

**Comment by tva10** [comment]
@SaintPeter I think that right way is getting real Style Declaration (as I written before. But my function need more test for each browser). For example it is possible to get that from `<style>` tag

**Comment by 4mifix** [comment]
Could it be that the CSS RULE read from top to bottom applies here? 
.thick-green-border {
   border-width: 10px;
   border-style: solid;
   border-color: green;
}

**Comment by SaintPeter** [comment]
I wonder if we can't test for a range?  I doubt that anyone is browsing at 50% < zoom < 150%, so maybe rather than testing for 10 specifically, we test for 
```
parseInt($('img.thick-green-border').css('border-top-width'),10) > 8 
  &&     
parseInt($('img.thick-green-border').css('border-top-width'),10) < 12
```

I tested on Chrome and it actually seems to flucutate between 9 and 10, but we can give ourselves some margin.

**Comment by ltegman** [comment]
@SaintPeter I'm a fan of this approach. It's not overly complicated and even if a user were to "incorrectly" pass this test when they'd actually put `11` instead of `10` their learning wouldn't have suffered at all since the exact value of the border wasn't point.

**Comment by jakhongirb** [comment]
Your img should be like the following
<img class="smaller-image thick-green-border" src="https://bit.ly/fcc-relaxing-cat">

instead of class="smaller-image" class="thick-green-border" 

I find it irritating that just a single space and the way one attributes a class to img can make a big difference.

Also, it's illogical and counterintuitive to make class applying two classes at the same time with just a space between them. I guess geeks that make css, html and programming languages have problems with logic and intuition. Because this is the issue I have ever time I start learning so-called eloquent and easy scripts.

**Comment by Kahldris** [comment]
This just happened to me - I couldn't clear it, it was driving me crazy. Cut and pasting back exact lines I had just written sometimes made one of the styles pass, it seemed so random. It wasn't on account of making my zoom 100% (I had it zoomed out to start but that was one of the first things I tried).

It eventually passed somehow by copy/pasting the example 3 styles from the left and changing the values, but this was identical to what I typed manually.

My Windows display scaling isn't at 100% (I'm using a Surface) so that might have contributed like a comment above said.

However when I go back to the challenge now I can't replicate it.

Please tell me this oddity won't happen in future challenges...

**Comment by gowthammurugan** [comment]
![screenshot 4](https://cloud.githubusercontent.com/assets/24809769/21538102/7205c4ee-cdbf-11e6-9cad-76de7a116047.png)
how to clear?

**Comment by felipealmeida28** [comment]
mine stayed like that, with two pictures and it worked perfectly
`
<link href="https://fonts.googleapis.com/css?family=Lobster" rel="stylesheet" type="text/css">
<style>
  .red-text {
    color: red;
  }

  h2 {
    font-family: Lobster, Monospace;
  }

  p {
    font-size: 16px;
    font-family: Monospace;
  }

  .smaller-image {
    width: 100px;
  }
  
  .thick-green-border { 
    border-color: green;
    border-style: 10px;
    border-style: solid;
    border-width: 10px;
    
  }
</style>

<h2 class="red-text">CatPhotoApp</h2>

<img class="thick-green-border" src="https://bit.ly/fcc-relaxing-cat" alt="borda verde" >

<img class="smaller-image" src="https://bit.ly/fcc-relaxing-cat" alt="A cute orange cat lying on its back. ">

<p class="red-text">Kitty ipsum dolor sit amet, shed everywhere shed everywhere stretching attack your ankles chase the red dot, hairball run catnip eat the grass sniff.</p>
<p class="red-text">Purr jump eat the grass rip the couch scratched sunbathe, shed everywhere rip the couch sleep in the sink fluffy fur catnip scratched.</p>
`
